internal/state/rollback.go

```diff
@@ -41,6 +41,10 @@ func Rollback(bs BlockStore, ss Store) (int64, []byte, error) {
 	if rollbackBlock == nil {
 		return -1, nil, fmt.Errorf("block at height %d not found", rollbackHeight)
 	}
+	lastBlock := bs.LoadBlockMeta(invalidState.LastBlockHeight)
```

Comment:
```suggestion
        // we also need to retrieve the latest block because the app hash and last results hash is only agreed upon in the following block
	lastBlock := bs.LoadBlockMeta(invalidState.LastBlockHeight)
```

Reply:
Do you think the `LastResultsHash` field should also be set to `latestBlock.Header.LastResultsHash`?

Reply:
I assume its under the same category of fields as app hash where it is only decided in the following block and hence the off by one error

Reply:
yeah, I think so too, changed.